src/components/Panel/Panel.test.tsx

```diff
@@ -1,129 +1,158 @@
 import { screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 
+import { usePanelQueryParams } from "panels/hooks";
 import { renderComponent } from "testing/utils";
 
 import Panel from "./Panel";
 
 describe("Panel", () => {
-  const url = "/foo?panel=share-model";
-
-  it("displays the title and content", async () => {
-    renderComponent(
-      <Panel panelClassName="test-panel" title="Test panel">
-        <>Test content</>
-      </Panel>,
-      { url }
-    );
+  const mockCheckCanClose = () => false;
+  const mockRenderComponent = ({
+    checkCanClose,
+  }: {
+    checkCanClose?: (
+      e: KeyboardEvent | React.MouseEvent<Element, MouseEvent>
+    ) => boolean;
+  }) => {
+    const MockPanelHeader = (): JSX.Element => {
+      const [, , handleRemovePanelQueryParams] = usePanelQueryParams<{
+        panel: null | string;
+      }>({ panel: null });
+
+      return (
+        <Panel
+          checkCanClose={checkCanClose}
+          panelClassName="test-panel"
+          title="Test panel"
+          onRemovePanelQueryParams={handleRemovePanelQueryParams}
+        >
+          <>Test content</>
+        </Panel>
+      );
+    };
+
+    return renderComponent(<MockPanelHeader />, {
+      url: "/foo?panel=share-model&externalParam=externalValue",
+    });
+  };
+
+  it("should render successfully", () => {
+    const {
+      result: { container },
+    } = mockRenderComponent({});
+
+    expect(container).toMatchSnapshot();
```

Comment:
This is fine, but we've generally been moving away from snapshots now as they're a lot less useful with RTL as it doesn't support shallow rendering.

Reply:
Fixed it in the latest commit and merged the PR.